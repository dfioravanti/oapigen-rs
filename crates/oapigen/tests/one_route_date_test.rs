use rstest::rstest;

#[rstest]
#[case("chrono", oapigen::DateTimeLibraries::Chrono)]
#[case("jiff", oapigen::DateTimeLibraries::Jiff)]
fn one_route_date_test(#[case] name: &str, #[case] library: oapigen::DateTimeLibraries) {
    let mut settings = insta::Settings::clone_current();
    settings.set_snapshot_suffix(name);

    let mut config = oapigen::Config::default();
    config.libraries.datetime = library;

    let f = std::fs::File::open("fixtures/one_route_date.yaml").unwrap();
    let spec: oas3::Spec = serde_yaml::from_reader(f).unwrap();

    let got = oapigen::spec_to_rust(&config, spec).unwrap();

    settings.bind(|| {
        insta::assert_snapshot!(got.to_string());
    });
}
