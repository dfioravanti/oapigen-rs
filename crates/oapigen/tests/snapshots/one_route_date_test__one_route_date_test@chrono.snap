---
source: crates/oapigen/tests/one_route_date_test.rs
expression: got.to_string()
---
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
///The requested date
#[derive(Debug, Deserialize, Serialize)]
struct GetDateResponse200(DateTime<Utc>);
